Content.Server/Chemistry/EntitySystems/ChemistrySystemHypospray.cs

```diff
@@ -30,23 +38,97 @@ private void OnSolutionChange(EntityUid uid, HyposprayComponent component, Solut
             Dirty(component);
         }
 
-        public void OnAfterInteract(EntityUid uid, HyposprayComponent comp, AfterInteractEvent args)
+        public void OnAfterInteract(EntityUid uid, HyposprayComponent component, AfterInteractEvent args)
         {
             if (!args.CanReach)
                 return;
 
             var target = args.Target;
             var user = args.User;
 
-            comp.TryDoInject(target, user);
+            TryDoInject(component, target, user);
         }
 
-        public void OnAttack(EntityUid uid, HyposprayComponent comp, MeleeHitEvent args)
+        public void OnAttack(EntityUid uid, HyposprayComponent component, MeleeHitEvent args)
         {
             if (!args.HitEntities.Any())
                 return;
 
-            comp.TryDoInject(args.HitEntities.First(), args.User);
+            TryDoInject(component, args.HitEntities.First(), args.User);
+        }
+
+        public bool TryDoInject(HyposprayComponent component, EntityUid? target, EntityUid user)
```

Comment:
public entitysystem API methods should follow the resolve structure, i.e.

```suggestion
        public bool TryDoInject(EntityUid uid, EntityUid? target, EntityUid user, HyposprayComponent? component=null)
        {
            if (!Resolve(uid, ref component)
                return;
```

https://docs.spacestation14.io/en/getting-started/conventions#public-api-method-signature